repo/fsrepo/migrations/retryfetcher.go

```diff
@@ -0,0 +1,43 @@
+package migrations
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+)
+
+type RetryFetcher struct {
+	Fetcher
+	maxRetries int
+}
+
+var _ Fetcher = (*RetryFetcher)(nil)
+
+func NewRetryFetcher(baseFetcher Fetcher, maxRetries int) *RetryFetcher {
+	return &RetryFetcher{Fetcher: baseFetcher, maxRetries: maxRetries}
+}
```

Comment:
@guseggert @lidel I added the retry fetcher from https://github.com/ipfs/ipfs-update/pull/158. This is really simplistic and likely we'll keep breaking this to add more complex backoff and retry strategies so maybe this doesn't belong here.

Similarly do we actually want to do this or do we just want to fallback to the next download strategy on the list if there's a problem (as opposed to in ipfs-update where we might just want to retry)?

Reply:
At least one retry is desirable IMO, stuff happens and it's a nicer experience to retry on transient issues.

I also just realized that we could have the same effect without having a special RetryFetcher by just putting the HTTP fetcher twice into the multifetcher.